src/main/webapp/src/app/app.component.ts

```diff
@@ -14,17 +16,32 @@ providedIn: 'root'
 
 export class AppComponent implements OnInit{
   title = 'Hidden Gems';
-  hiddenGems! : HiddenGem[];
+  hiddenGems!: HiddenGem[];
+  location = {} as Location;
+  previousLocation = {} as Location;
 
-  constructor(private hiddenGemService: HiddenGemService) { }
+  constructor(private hiddenGemService: HiddenGemService, private locationService: LocationService) {
+    this.locationService.getLocation
+      .subscribe(location => {
+        this.location = location;
+      })
+  }
 
   ngOnInit() {
     this.getRankedHiddenGems();
+    this.previousLocation = this.location;
+  }
+
+  ngDoCheck() {
+    if (this.location !== this.previousLocation) {
```

Comment:
As far as I can tell we only load the location once.
Is this planning for future features or was this because getting the location takes a little while?
Either way could you add a comment explaining why it's necessary.

Reply:
It's because the location is initially set as the default location (Sydney) when the app is initialised. The user needs to click on using either their location or Sydney or Adelaide, so this code detects the change of location and update it before retrieving the hidden gems. I added a comment for it.